### PR TITLE
Pullrequest contact picker

### PR DESCRIPTION
**Pull request by BenSartor**

### CODE REVIEW

File: app/src/main/java/org/simlar/widgets/NoContactPermissionFragment.java [hunk 1]
```diff
@@ -66,5 +75,28 @@ private void requestContactPermissionsClicked()
 	private void callContactClicked()
 	{
 		Lg.i("callContactClicked");
+
+		final Intent intent = new Intent(Intent.ACTION_PICK, ContactsContract.CommonDataKinds.Phone.CONTENT_URI);
+		startActivityForResult(intent, PICK_CONTACT);
+	}
+
+	@Override
+	public void onActivityResult(final int requestCode, final int resultCode, final Intent data) {
```
Comment (jritzerfeld) [inline app/src/main/java/org/simlar/widgets/NoContactPermissionFragment.java]: AFAIK, the '{' is placed in an extra line in our code.

File: app/src/main/java/org/simlar/widgets/NoContactPermissionFragment.java [hunk 2]
```diff
@@ -66,5 +75,28 @@ private void requestContactPermissionsClicked()
...
+		Lg.i("onActivityResult");
+
+		if (requestCode == PICK_CONTACT && resultCode == RESULT_OK) {
+			final Uri contactUri = data.getData();
+			final Cursor cursor = getActivity().getContentResolver().query(contactUri, null, null, null, null);
+			if (cursor != null) {
```
Comment (jritzerfeld) [inline app/src/main/java/org/simlar/widgets/NoContactPermissionFragment.java]: Personally, I would reduce indentation by
`if (cursor == null) {`
`    return;`
`}`

File: app/src/main/java/org/simlar/helper/PermissionsHelper.java [hunk 3]
```diff
@@ -169,6 +170,11 @@ public void onDismiss(final DialogInterface dialog)
 				.create().show();
 	}
 
+	public static void requestContactPermission(final Activity activity)
+	{
+		requestPermissions(activity, new HashSet<>(Arrays.asList(new Type[]{ Type.CONTACTS })));
```
Comment (jritzerfeld) [inline app/src/main/java/org/simlar/helper/PermissionsHelper.java]: Maybe `requestPermissions(activity, Collections.singelton(Type.CONTACTS));`?

File: app/src/main/java/org/simlar/utils/Util.java [hunk 4]
```diff
@@ -104,6 +106,15 @@ public static Drawable getDrawableCompatible(final Resources resources, final in
 		}
 	}
 
+	public static Spanned fromHtml(final String string) {
```
Comment (jritzerfeld) [inline app/src/main/java/org/simlar/utils/Util.java]: `{` in a separate line, please.

File: app/src/main/java/org/simlar/widgets/NoContactPermissionFragment.java (hunk 2 above)
Comment (BenSartor) [inline app/src/main/java/org/simlar/widgets/NoContactPermissionFragment.java]: I was thinking about:
if (requestCode != PICK_CONTACT && resultCode != RESULT_OK) {
  return;
}

Lets talk about it in our next Simlar call.

Comment (jritzerfeld) [inline app/src/main/java/org/simlar/widgets/NoContactPermissionFragment.java]: In addition: Good idea and with `||` please.

**Pull request merged**